mailstore-connector/src/com/axonivy/connector/mailstore/MailStoreService.java

```diff
@@ -362,9 +385,18 @@ private MessageIterator(String storeName, String srcFolderName, String dstFolder
 				this.delete = delete;
 				store = MailStoreService.openStore(storeName);
 				srcFolder = MailStoreService.openFolder(store, srcFolderName, Folder.READ_WRITE);
-				if(StringUtils.isNotBlank(dstFolderName)) {
-					dstFolder = MailStoreService.openFolder(store, dstFolderName, Folder.READ_WRITE);
+				
+				if(CollectionUtils.isNotEmpty(dstFolderNames)) {
+					dstFolderMap = new LinkedHashMap<>();

```

Comment:
The `dstFolderMap` should be initialize at the beginning to avoid NPE exception if method `handledMessage` is called when no `dstFolderNames` has been defined,